### PR TITLE
Update GetAllRevshare to handle liquidations

### PR DESCRIPTION
**Pull request by affanv14**
### Changelist
[Failure error logs](https://app.datadoghq.com/logs?query=issue.id%3Ab70fa6d4-7d6d-11ef-909c-da7ad0900002%20&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&from_ts=1727507763000&to_ts=1727512431724&live=false) when going through liquidations since maker and taker fees are 0. This handles that case

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Enhanced revenue sharing functionality with improved checks for net fees.
	- Refined logic for calculating affiliate revenue shares based on specified thresholds.
	- Clarified error messages related to affiliate fees.

- **Bug Fixes**
	- Improved error handling for revenue share calculations to prevent exceeding defined limits.

- **Tests**
	- Added new test cases to validate behavior with zero fees and refined existing tests for accuracy in revenue share calculations.
	- Standardized setup functions for consistency across test scenarios.

### CODE REVIEW

**Review by teddyding — APPROVED** [review]

File: protocol/x/revshare/keeper/revshare.go [hunk 1]
```diff
@@ -175,6 +175,11 @@ func (k Keeper) GetAllRevShares(
...
 	netFees := big.NewInt(0).Add(takerFees, makerFees)
 
+	// net fees is 0 in case of liquidations where maker and taker fees are 0
+	if netFees.Sign() == 0 {
+		return types.RevSharesForFill{}, nil
+	}
+
 	affiliateRevShares, affiliateFeesShared, err := k.getAffiliateRevShares(ctx, fill, affiliatesWhitelistMap)
 	if err != nil {
 		return types.RevSharesForFill{}, err
```
Comment (teddyding) [inline protocol/x/revshare/keeper/revshare.go:185]: Comment on this [line](https://github.com/dydxprotocol/v4-chain/pull/2413/files#diff-1203e511f1b1da0901525ce89b0abb29818a2cc88b7f5e916f0734df1a7d9de6R188) below: error message `affiliate fees shared exceeds net fees"` is not accurate since we return error when `netFeesSubAffiliateFeesShared = 0` as well

File: protocol/x/revshare/keeper/revshare.go [hunk 2]
```diff
@@ -175,6 +175,11 @@ func (k Keeper) GetAllRevShares(
 	makerFees := fill.MakerFeeQuoteQuantums
 	netFees := big.NewInt(0).Add(takerFees, makerFees)
 
+	// net fees is 0 in case of liquidations where maker and taker fees are 0
```
Comment (teddyding) [inline protocol/x/revshare/keeper/revshare.go:178]: This comment is not general enough. Net fees can be zero if `takerFee > 0` and `makerFee < 0`. It should just say "when net fee is zero, not rev share is generated from the fill'

File: protocol/x/revshare/keeper/revshare_test.go [hunk 3]
```diff
@@ -726,6 +726,48 @@ func TestKeeper_GetAllRevShares_Valid(t *testing.T) {
 				affiliatesKeeper *affiliateskeeper.Keeper) {
 			},
 		},
+		{
```
Comment (teddyding) [inline protocol/x/revshare/keeper/revshare_test.go:729]: From thread, could we also add a unit test with positive maker fees and zero taker fee?

Comment (affanv14) [reply protocol/x/revshare/keeper/revshare_test.go:729]: good catch!

File: protocol/x/revshare/keeper/revshare.go [hunk 4]
```diff
@@ -238,7 +238,8 @@ func (k Keeper) getAffiliateRevShares(
 ) ([]types.RevShare, *big.Int, error) {
 	takerAddr := fill.TakerAddr
 	takerFee := fill.TakerFeeQuoteQuantums
-	if fill.MonthlyRollingTakerVolumeQuantums >= types.MaxReferee30dVolumeForAffiliateShareQuantums {
+	if fill.MonthlyRollingTakerVolumeQuantums >= types.MaxReferee30dVolumeForAffiliateShareQuantums ||
```
Comment (affanv14) [inline protocol/x/revshare/keeper/revshare.go:241]: Better to stay consistent with the 0 net fee use case here instead of return a 0 fee revshare

**Review by jayy04 — APPROVED** [review]

File: protocol/x/revshare/types/errors.go [hunk 5]
```diff
@@ -36,9 +36,9 @@ var (
 		6,
 		"total fees shared exceeds net fees",
 	)
-	ErrAffiliateFeesSharedExceedsNetFees = errorsmod.Register(
+	ErrAffiliateFeesSharedGreaterThanOrEqualToNetFees = errorsmod.Register(
 		ModuleName,
 		7,
-		"affiliate fees shared exceeds net fees",
+		"net fees minus affiliate fee share is not larger than zero",
```
Comment (teddyding) [inline protocol/x/revshare/types/errors.go:42]: Nit: var name and err message is saying the same thing but differently:

```
AffiliateFee >= netFee
netFee - AffiliateFee <= 0
```

Let's make them consistent for readability

**Review by teddyding — APPROVED** [review]

**Comment by affanv14** [comment]
@mergifyio backport release/protocol/v7.x
